tensorflow/lite/kernels/test_delegate_providers.cc

```diff
@@ -29,6 +29,8 @@ constexpr char KernelTestDelegateProviders::kUseSimpleAllocator[];
 
 KernelTestDelegateProviders::KernelTestDelegateProviders()
     : delegate_list_util_(&params_) {
+  TFLITE_TOOLS_CHECK(delegate_list_util_.GetCount() > 0)
```

Comment:
If there is no delegate available, calling _delegate_list_util_.AppendCmdlineFlags(&flags);_ (line 46) appends no command line flags. Since at least _--help_ flag is expected to be available (line 49), the condition returns false and an error is reported (specifically: "help was not found", as implemented in _tensorflow/lite/tools/tool_params.cc:53_). So this check aims to provide the user with more instructive/helpful error message, should this happen again in the future.